users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -401,6 +616,203 @@ def process_args(args: Dict[str, Any]):
     return report
 
 
+def run_scf_audio_perturbation_from_checkpoint():
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/ctc/feat/"
+
+    _, nn_system = run_scf_baseline()
+
+    (
+        returnn_datasets,
+        rasr_loss_corpus_path,
+        rasr_loss_corpus_segments,
+        rasr_loss_lexicon_path,
+        dev_corpora,
+    ) = get_datasets(use_multi_proc_dataset=True, pre_process=CodeWrapper("audio_perturb_runner.run"))
+    returnn_args = {
+        "batch_size": 5000,
+        "rasr_binary_path": RASR_BINARY_PATH,
+        "rasr_loss_corpus_path": rasr_loss_corpus_path,
+        "rasr_loss_corpus_segments": rasr_loss_corpus_segments,
+        "rasr_loss_lexicon_path": rasr_loss_lexicon_path,
+        "datasets": returnn_datasets,
+        "conformer_type": "wei",
+        "audio_perturbation": True,
+        "use_multi_proc_dataset": True,
+        "specaug_old": {"max_feature": 15},
+    }
+    # lr_args={                      Since we use a checkpoint the correct lr_args would be these, but tests showed that restarting the lr schedule is better
+    #     "peak_lr": 4e-4,
+    #     "start_lr": 6.2668056e-05,
+    #     "end_lr": 1e-5,
+    #     "increase_epochs": 156,
+    #     "decrease_epochs": 180,
+    #     "final_epochs": 0,
+    # },
+    lr_args = {
+        "peak_lr": 4e-4,
+        "start_lr": 1.325e-05,
+        "end_lr": 1e-5,
+        "increase_epochs": 180,
+        "decrease_epochs": 180,
+        "final_epochs": 0,
+    }
+
+    perturbation_args = [
```

Comment:
I do not really understand the question. The perturbation experiments all use the same checkpoint to start. The checkpoint is from the baseline and does not do any perturbation.

Reply:
My question was: Do you run all these experiments (quite a number) both 1) from scratch and 2) starting from the checkpoint? Or always starting from the checkpoint?

Reply:
Always from the checkpoint. It is still a lot but I save the first 4 epochs for each training. Starting perturbation at the beginning without the checkpoint is unstable and cannot guarantee that all original training data were used.

Reply:
I'm wondering why you have `run_scf_audio_perturbation` and `run_scf_audio_perturbation_from_checkpoint` now if you always want to run from the checkpoint. In the `py` function (acts as `main`), you call both which indicates to me that both should be run.

Reply:
Now that bigger batch sizes are possible, I could try to perform perturbation from the start. But I do not think it is worth the time and compute. I would remove the  `run_scf_audio_perturbation` function and would do the same for log mel since I mostly need them as comparison.